test/correctness/bounds_of_shift.cpp

```diff
@@ -0,0 +1,43 @@
+#include "Halide.h"
+#include <stdio.h>
+
+using namespace Halide;
+
+void check(Func f, ImageParam in, int min, int extent) {
+    Buffer<int> output(12345);
+    output.set_min(-1234);
+
+    in.reset();
+    f.infer_input_bounds(output);
+    Buffer<int> im = in.get();
+
+    if (im.extent(0) != extent || im.min(0) != min) {
+        printf("Inferred size was [%d, %d] instead of [%d, %d]\n",
+               im.min(0), im.extent(0), min, extent);
+        exit(-1);
+    }
+}
+
+int main(int argc, char **argv) {
+    ImageParam input(Int(32), 1);
+    Var x;
+    Func f1 = lambda(x, input(cast<int8_t>(x) << 2));
```

Comment:
Constant shifts get rewritten to multiplies/divides, so I think you need a Param in this test, e.g.: https://github.com/halide/Halide/blob/master/test/correctness/param.cpp#L13